pytorch_lightning/callbacks/progress/base.py

```diff
@@ -105,12 +124,11 @@ def total_val_batches(self) -> Union[int, float]:
         Use this to set the total number of iterations in the progress bar. Can return ``inf`` if the validation
         dataloader is of infinite size.
         """
-        total_val_batches = 0
-        if self.trainer.enable_validation:
-            is_val_epoch = (self.trainer.current_epoch + 1) % self.trainer.check_val_every_n_epoch == 0
-            total_val_batches = sum(self.trainer.num_val_batches) if is_val_epoch else 0
+        assert self._current_eval_dataloader_idx is not None
+        if self.trainer.sanity_checking:
+            return self.trainer.num_sanity_val_batches[self._current_eval_dataloader_idx]
 
-        return total_val_batches
+        return self.trainer.num_val_batches[self._current_eval_dataloader_idx]
```

Comment:
total_meth_batches returns the num_batches for the current dataloader since we now create a separate bar for each dataloader.